### PR TITLE
Feature/deprecate v2 api

### PR DESCRIPTION
**Pull request by jasonpaige**
**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/draftstore/endpoint/v2/DraftStoreEndpoint.java [hunk 1]
```diff
@@ -27,6 +28,12 @@
 import static org.springframework.web.bind.annotation.RequestMethod.POST;
 import static uk.gov.hmcts.reform.draftstore.domain.SaveStatus.Updated;
 
+@APIDeprecated(
+    name = "/api/v2/draft endpoint",
+    expiryDate = "2018-01-31",
+    docLink = "https://github.com/hmcts/draft-store/blob/master/README.md",
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/draftstore/endpoint/v2/DraftStoreEndpoint.java]: or https://github.com/hmcts/draft-store#encryption-feature-in-v3

File: README.md [hunk 2]
```diff
@@ -5,6 +5,14 @@
...
 The Draft store micro-service provides a ’save and resume’ feature for reform applications. 
 It stores a draft json document against the userId provided.
+
+## Encryption feature in v3
+A new version of the API has been created to meet [point 6 of the NCSC Security Design Principles: Reducing the impact 
+of compromise](https://www.ncsc.gov.uk/guidance/design-principles-reducing-impact-compromise). 
+The current API allows for both client to _optionally_ supply an encryption key. This optionality will be revoked in the 
+future to ensure all partially completed form data being stored by service teams is encrypted per user.
+
+For this reason, version 2 of the draft-store API is also now deprecated and will be removed on 31st Janurary 2018.
```
Comment (rkondratowicz) [inline README.md]: typo in 'January'

File: src/main/java/uk/gov/hmcts/reform/draftstore/endpoint/v2/DraftStoreEndpoint.java [hunk 3]
```diff
@@ -27,6 +28,12 @@
 import static org.springframework.web.bind.annotation.RequestMethod.POST;
 import static uk.gov.hmcts.reform.draftstore.domain.SaveStatus.Updated;
 
+@APIDeprecated(
+    name = "/api/v2/draft endpoint",
+    expiryDate = "2018-01-31",
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/draftstore/endpoint/v2/DraftStoreEndpoint.java]: Can we make it `2017-11-31` and then extend if needed?

Comment (jasonpaige) [reply src/main/java/uk/gov/hmcts/reform/draftstore/endpoint/v2/DraftStoreEndpoint.java]: doesn't really make sense to go that way around. we'll set a long date and then reduce it if we can. getting 2 deployment slots in November given activity on the program and no CD at the moment seems unlikely

**Pull request merged**